src/gui/dbsettings/DatabaseSettingsWidgetGeneral.cpp

```diff
@@ -130,6 +138,16 @@ bool DatabaseSettingsWidgetGeneral::save()
         truncate = true;
     }
 
+    int autosaveDelay;
+    if (m_ui->autosaveDelayCheckBox->isChecked()) {
+        autosaveDelay = m_ui->autosaveDelaySpinBox->value();
+    } else {
+        autosaveDelay = 0;
+    }
+    if (autosaveDelay != meta->autosaveDelay()) {
```

Comment:
I don't think this condition is useful. What will happen if we set the current value always?

Reply:
i agree, fixed in commit [4438e9f](https://github.com/keepassxreboot/keepassxc/pull/9100/commits/4438e9fc992dbdba424555be631451183ed832b8)
I think the same could be done for lines 119-139